Source/Security/SDKeychain.m

```diff
@@ -127,6 +127,79 @@ + (BOOL)setString:(NSString*)string forKey:(NSString*)key serviceName:(NSString
     return [self setString:string forKey:key serviceName:serviceName synchronizeViaiCloud:NO accessGroup:nil];
 }
 
++ (BOOL)isPasswordAvailableInBackgroundForUsername:(NSString *)username andServiceName:(NSString *)serviceName synchronizeViaiCloud:(BOOL)synchronizeViaiCloud accessGroup:(NSString *)accessGroup error:(NSError **)error
+{
+    BOOL isAvailableInBackground = NO;
+    if (!username || !serviceName)
+    {
+        if (error != nil)
+            *error = [NSError errorWithDomain:SDKeychainErrorDomain code:-2000 userInfo:nil];
+        return NO;
+    }
+
+    if (error != nil)
+        *error = nil;
+
+    // Set up a query dictionary with the base query attributes: item type (generic), username, and service
+    NSMutableDictionary *query = [NSMutableDictionary dictionaryWithDictionary: @{
+                                                                                  (__bridge NSString *)kSecClass: (__bridge NSString *)kSecClassGenericPassword,
+                                                                                  (__bridge id)kSecAttrAccount: username,
+                                                                                  (__bridge id)kSecAttrService: serviceName
+                                                                                  }];
+
+#if !TARGET_IPHONE_SIMULATOR
+    if (accessGroup) {
+        query[(__bridge id)kSecAttrAccessGroup] = [NSString stringWithFormat:@"%@%@", [SDKeychain bundleSeedID], accessGroup];
+    }
+#endif
+
+    if (synchronizeViaiCloud) {
+        query[(__bridge id)(kSecAttrSynchronizable)] = @YES;
+    }
+
+    /*
+     NSArray *keys = [[NSArray alloc] initWithObjects:(__bridge NSString *)kSecClass, kSecAttrAccount, kSecAttrService, nil];
+     NSArray *objects = [[NSArray alloc] initWithObjects:(__bridge NSString *)kSecClassGenericPassword, username, serviceName, nil];
+
+     NSMutableDictionary *query = [[NSMutableDictionary alloc] initWithObjects:objects forKeys:keys];
+     */
+
```

Comment:
yeah good call, this is a copy/paste of a similar method. I'll clean this one up